pilot/pkg/xds/cds_test.go

```diff
@@ -170,36 +172,35 @@ func TestSAN(t *testing.T) {
 			name:    "Kubernetes service and EDS ServiceEntry",
```

Comment:
This case and the next 4 legitimately are changing behavior here.

Before: no sidecar. Service conflicts, so we pick the k8s service over the SE

After: simulated sidecar. Service conflicts, so we pick the same-namespace SE over the k8s-service in another namespace.

Note the 'after' behavior here occurs today if you have a Sidecar object.